api/src/view_function.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    accept_type::AcceptType,
+    failpoint::fail_point_poem,
+    response::{
+        BadRequestError, BasicErrorWith404, BasicResponse, BasicResponseStatus, BasicResultWith404,
+    },
+    ApiTags, Context,
+};
+use anyhow::Context as AnyhowContext;
+use aptos_api_types::{
+    verify_function_identifier, verify_module_identifier, Address, AptosErrorCode,
+    IdentifierWrapper, MoveValue, ViewRequest, U64,
+};
+use aptos_vm::AptosVM;
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{ModuleId, TypeTag},
+};
+use poem_openapi::{
+    param::{Path, Query},
+    payload::Json,
+    OpenApi,
+};
+use std::{convert::TryInto, sync::Arc};
+
+/// API for executing Move view function.
+pub struct ViewFunctionApi {
+    pub context: Arc<Context>,
+}
+
+#[OpenApi]
+impl ViewFunctionApi {
+    /// Execute view function of the module
+    ///
+    /// Execute the Move function with the given paramters and return the execution result.
+    ///
+    /// The Aptos nodes prune account state history, via a configurable time window.
+    /// If the requested ledger version has been pruned, the server responds with a 410.
+    #[oai(
+        path = "/view/:address/module/:module_name/function/:function_name",
```

Comment:
Pull all of this out of the path, and put it in the body to be similar to `simulate_transaction`